controllers/accounts_test.go

```diff
@@ -259,6 +269,183 @@ func (suite *AccountsTestSuite) TestAccountDeletionEndpointEnabled() {
+		GenerateQR: true,
+	})
+	initReq.Header.Set("Authorization", "Bearer "+token)
+	initResp := util.ExecuteTestRequest(initReq, suite.router)
+	suite.Equal(http.StatusOK, initResp.Code)
+
+	var initParsedResp controllers.TwoFAInitResponse
+	util.DecodeJSONTestResponse(suite.T(), initResp.Body, &initParsedResp)
+	suite.Require().NotEmpty(initParsedResp.URL)
+	suite.Require().NotNil(initParsedResp.QRCode)
+	suite.NotEmpty(*initParsedResp.QRCode)
+
+	// Parse the TOTP URL to extract the secret
+	key, err := otp.NewKeyFromURL(initParsedResp.URL)
+	suite.Require().NoError(err)
+	suite.Equal("Brave Account", key.Issuer())
+	suite.Equal("test@example.com", key.AccountName())
+	suite.True(strings.HasPrefix(initParsedResp.URL, "otpauth://totp/Brave%20Account"))
+
+	// Test finalizing TOTP setup with invalid code
+	invalidCode := "000000"
+	finalizeReq := util.CreateJSONTestRequest("/v2/accounts/2fa/totp/finalize", controllers.TwoFAFinalizeRequest{
+		Code: invalidCode,
+	})
+	finalizeReq.Header.Set("Authorization", "Bearer "+token)
+	finalizeResp := util.ExecuteTestRequest(finalizeReq, suite.router)
+	suite.Equal(http.StatusBadRequest, finalizeResp.Code)
+	util.AssertErrorResponseCode(suite.T(), finalizeResp, util.ErrBadTOTPCode.Code)
+
+	// Test finalizing TOTP setup with valid code
+	validCode, err := totp.GenerateCode(key.Secret(), time.Now().UTC())
+	suite.Require().NoError(err)
+
+	finalizeReq = util.CreateJSONTestRequest("/v2/accounts/2fa/totp/finalize", controllers.TwoFAFinalizeRequest{
+		Code: validCode,
+	})
+	finalizeReq.Header.Set("Authorization", "Bearer "+token)
+	finalizeResp = util.ExecuteTestRequest(finalizeReq, suite.router)
+	suite.Equal(http.StatusOK, finalizeResp.Code)
+
+	var finalizeParsedResp controllers.TwoFAFinalizeResponse
+	util.DecodeJSONTestResponse(suite.T(), finalizeResp.Body, &finalizeParsedResp)
+	suite.Require().NotNil(finalizeParsedResp.RecoveryKey)
+	suite.NotEmpty(*finalizeParsedResp.RecoveryKey)
+
+	// Verify 2FA is now enabled
+	updatedAccount, err := suite.ds.GetAccount(nil, account.Email)
+	suite.Require().NoError(err)
+	suite.True(updatedAccount.IsTwoFAEnabled())
+	suite.NotNil(updatedAccount.RecoveryKeyHash)
+
+	// Test initializing TOTP when it's already enabled
+	initReq = util.CreateJSONTestRequest("/v2/accounts/2fa/totp/init", controllers.TwoFAInitRequest{
+		GenerateQR: true,
+	})
+	initReq.Header.Set("Authorization", "Bearer "+token)
+	initResp = util.ExecuteTestRequest(initReq, suite.router)
+	suite.Equal(http.StatusBadRequest, initResp.Code)
+	util.AssertErrorResponseCode(suite.T(), initResp, util.ErrTOTPAlreadyEnabled.Code)
+}
+
+func (suite *AccountsTestSuite) TestDisableTOTP() {
+	token, account := suite.createAuthSession()
+
+	// Enable 2FA first
+	err := suite.ds.SetTOTPSetting(account.ID, true)
+	suite.Require().NoError(err)
+
+	// Generate and store a recovery key
+	recoveryKey := "test-recovery-key-12345"
```

Comment:
Let's use a properly-formatted one here so that we can validate it in `SetRecoveryKey()`.